Documentation for .NET Core Uninstall Tool (#15288)
dotnet/docs: docs/core/additional-tools/dotnet-core-uninstall.md

---
title: .NET Core Uninstall Tool
description: An overview of the .NET Core Uninstall Tool, a guided tool that enables the controlled clean-up of .NET Core SDKs and Runtimes.
author: yuchong-pan
ms.date: 08/06/2019
---
# .NET Core Uninstall Tool

The [.NET Core Uninstall Tool](https://dotnet.microsoft.com/download/dotnet-core/uninstall-tool) (`dotnet-core-uninstall`) lets you clean-up .NET Core SDKs and Runtimes on a system such that only the desired versions remain. A collection of options is available to specify which versions you want to uninstall.

The tool supports Windows and macOS. Linux is currently not supported.

Note that the Windows version of this tool can only uninstall .NET Core SDKs and Runtimes that were installed using .NET Core SDK or Runtime installers. Prior to Visual Studio 2019 Update 16.3, the Visual Studio installer called the .NET Core SDK installer and these can be uninstalled with the tool. The `dotnet-core-uninstall list` command displays which SDKs can be uninstalled with the tool.  

## How to use the tool

1. Download and install the tool.

> 2. Display installed .NET Core SDKs and Runtimes that can be removed by this tool.

Reviewer: ```suggestion
2. Display installed .NET Core SDKs and Runtimes that can be removed by this tool using `dotnet-core-uninstall list`.
```

> 3. Do a dry run to ensure the right things will be uninstalled, and uninstall .NET Core SDKs and Runtimes by running the tool with Admin privileges.

Reviewer: ```suggestion
3. Use `dotnet-core-uninstall dry-run` to ensure the right things will be uninstalled.
4. Use `dotnet-core-uninstall remove` to uninstall .NET Core SDKs and Runtimes. This must be run with Admin privileges.
```

```

#### Options

### [Windows](#tab/windows)

* **`--aspnet-runtime`**

  List ASP.NET Core Runtimes.

* **`--hosting-bundle`**

  List .NET Core Runtime & Hosting Bundles.

* **`--runtime`**

  List .NET Core Runtimes.

* **`--sdk`**

  List .NET Core SDKs.

* **`-v, --verbosity <LEVEL>`**

  Set the verbosity level. Allowed values are `q[uiet]`, `m[inimal]`, `n[ormal]`, `d[etailed]`, and `diag[nostic]`.

* **`--x64`**

  List x64 .NET Core SDKs or Runtimes.

* **`--x86`**

  List x86 .NET Core SDKs or Runtimes.

### [macOS](#tab/macos)

* **`--runtime`**

  List .NET Core Runtimes.

* **`--sdk`**

  List .NET Core SDKs.

* **`-v, --verbosity <LEVEL>`**

  Set the verbosity level. Allowed values are `q[uiet]`, `m[inimal]`, `n[ormal]`, `d[etailed]`, and `diag[nostic]`.
  
---

#### Examples

* List all .NET Core SDKs and Runtimes that can be removed with this tool:

  ``` console
  dotnet-core-uninstall list
  ```

* List all x64 .NET Core SDKs and Runtimes:

  ``` console
  dotnet-core-uninstall list --x64
  ```

* List all x86 .NET Core SDKs:

  ``` console
  dotnet-core-uninstall list --sdk --x86
  ```

### Uninstalling .NET Core SDKs and Runtimes

> `dotnet-core-uninstall dry-run` and `dotnet-core-uninstall whatif` display the .NET Core SDKs and Runtimes that will be removed based on the options provided without performing the uninstall. These commands are equivalent.

Reviewer: ```suggestion
`dotnet-core-uninstall dry-run` and `dotnet-core-uninstall whatif` display the .NET Core SDKs and Runtimes that will be removed based on the options provided without performing the uninstall. These commands are synonyms.
```

> Since this is a destructive tool, it is **highly** recommended that you run one of these commands before the remove command to determine the .NET Core SDKs and Runtimes that will be removed. [This article explains which SDKs and Runtimes are safe to remove.](/versions/remove-runtime-sdk-versions#should-i-remove-a-version)

Reviewer: ```suggestion
Since this is a destructive tool, it is **highly** recommended that you do a dry run before the running the remove command. This will tell you what .NET Core SDKs and Runtimes that will be removed. [This article explains which SDKs and Runtimes are safe to remove.](/versions/remove-runtime-sdk-versions#should-i-remove-a-version)
```

---
title: .NET Core Uninstall Tool
description: An overview of the .NET Core Uninstall Tool, a guided tool that enables the controlled clean-up of .NET Core SDKs and runtimes.
author: yuchong-pan
ms.date: 10/31/2019
---
# .NET Core Uninstall Tool

The [.NET Core Uninstall Tool](https://github.com/dotnet/cli-lab/releases) (`dotnet-core-uninstall`) lets you clean up .NET Core SDKs and runtimes on a system such that only the desired versions remain. A collection of options is available to specify which versions you want to uninstall.

The tool supports Windows and macOS. Linux is currently not supported.

> Note that the Windows version of this tool can only uninstall .NET Core SDKs and runtimes that were installed using .NET Core SDK and runtime installers. Prior to Visual Studio 2019 Update 16.3, the Visual Studio installer called the .NET Core SDK installer and these versions can be uninstalled with the tool. The `dotnet-core-uninstall list` command displays which SDKs can be uninstalled with the tool.

Reviewer 2: What happens starting with 16.3?

Author: @wli3 would know more than me

Reviewer 2: @wli3 can you please comment?

Reviewer 3: After that Visual Studio is in charge of its own copy. That means Visual Studio will bring a copy of dotnet core, and the user cannot uninstall it from "add remove programs". And Visual Studio will also update that copy in VS update. It will also go away when VS is uninstalled

## Install the tool

You can download the .NET Core Uninstall Tool from [GitHub Releases](https://github.com/dotnet/cli-lab/releases).

> [!NOTE]
> The tool requires elevation to uninstall .NET Core SDKs and runtimes. Therefore, it should be installed in a write-protected directory such as *C:\Program Files* on Windows or */usr/local/bin* on macOS. See also [Elevated access for dotnet commands](../tools/elevated-access.md). Detailed installation instructions are available on the [GitHub Releases page](https://github.com/dotnet/cli-lab/releases).

## How to use the tool

> 1. Display installed .NET Core SDKs and runtimes that can be removed by this tool using `dotnet-core-uninstall list`.

Reviewer 2: it felt weird to have install and uninstall steps in the how to use the tool guide.

Also, I'm wondering if each one of these items should be simplified and be a link to the section below or if we should simply skip and have the steps directly in the header such as in https://docs.microsoft.com/en-us/visualstudio/install/install-visual-studio?view=vs-2019

I'm also thinking that usage instructions and command syntax should be broken down into different sections.

Author: Having the steps directly in the header would make sense if we're splitting the options for `dotnet-core-uninstall dry-run` and `dotnet-core-uninstall remove` into two different sections. Is that what you mean by braking down the usage instructions or did you have something else in mind?

Reviewer 2: I think install, use, and uninstall should be separate instructions instead of having install and uninstall as steps of how to use the tool - that seems wrong. I tried to make that more clear in the latest updates - see what you think! But I think what I'm also struggling with on this particular article is that is a mix of reference (with the commands syntax) and how to. 
I see two options:
- We merge as is (if you're ok with my changes) and watch for customer's feedback.
- We separate the commands syntax and the prescriptive guidance into different sections (or even different articles). We could even merge this and then I open a new PR with a proposal on top of this.

Author: It's up to you, but the changes you made look good to me and I am okay with merging this if @KathleenDollard is.

2. Use `dotnet-core-uninstall dry-run` to ensure the right SDKs and runtimes will be uninstalled.
3. Use `dotnet-core-uninstall remove` to uninstall .NET Core SDKs and runtimes. This command must be run with Admin privileges.
4. Delete the `NuGetFallbackFolder` if you no longer need it (optional).

### Display installed .NET Core SDKs and runtimes

Use the `dotnet-core-uninstall list` command lists installed .NET Core SDKs and runtimes that can be removed with this tool. Some SDKs and runtimes may be required by Visual Studio and they're displayed with a note of why it isn't recommended to uninstall them.

> On Windows, the tool can only uninstall SDKs and runtimes that were installed using the .NET Core SDK and runtime installer, or the Visual Studio installer prior to Visual Studio 2019 16.3. On macOS, the tool can only uninstall SDKs and runtimes located in the */usr/local/share/dotnet* folder. This tool doesn't work on Linux. Because of these limitations, the tool may not be able to uninstall all of the .NET Core SDKs and runtimes on your machine. You can use the `dotnet --info` command to find all of the .NET Core SDKs and runtimes available, including those SDKs and runtimes that this tool can't remove.

Reviewer 2: this info about 16.3 and that it doesn't work on Linux was already mentioned in the intro. We should avoid repeating the same info so we should choose where it makes more sense to have this.

Author: Probably makes sense to have it in the intro then.
